Kentico.Kontent.Management.Tests/Unit/ManagementClientTests/CollectionTests.cs

```diff
@@ -0,0 +1,165 @@
+﻿using FluentAssertions;
+using Kentico.Kontent.Management.Models.Collections;
+using Kentico.Kontent.Management.Models.Collections.Patch;
+using Kentico.Kontent.Management.Models.Shared;
+using Kentico.Kontent.Management.Tests.Unit.Base;
+using System;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace Kentico.Kontent.Management.Tests.Unit.ManagementClientTests
+{
+    public class CollectionTests : IClassFixture<FileSystemFixture>
+    {
+        private FileSystemFixture _fileSystemFixture;
+
+        public CollectionTests(FileSystemFixture fileSystemFixture)
+        {
+            _fileSystemFixture = fileSystemFixture;
+            _fileSystemFixture.SetSubFolder("Collection");
+        }
+
+        [Fact]
+        public async Task ListCollections_ListsCollections()
+        {
+            var client = _fileSystemFixture.CreateMockClientWithResponse("Collections.json");
+
+            var expectedItems = _fileSystemFixture.GetExpectedResponse<CollectionsModel>("Collections.json");
+
+            var response = await client.ListCollectionsAsync();
+
+            response.Should().BeEquivalentTo(expectedItems);
+        }
+
+        
+        [Fact]
+        public async void ModifyCollection_Remove_ById_RemovesCollection()
+        {
+            var client = _fileSystemFixture.CreateMockClientWithoutResposne();
```

Comment:
So this test is for delete, of removing the identifier? 

The original test `Kentico.Kontent.Management.Tests.ManagementClientTests.ModifyCollection_Remove_ById_RemovesCollection` was testing that exception was thrown right?

Reply:
Not really, It worked against live endpoint like this: 
1. It creates a collection
2. It removes it by its id.
3. It tries to remove it once again by its id but now it does not exist so we expect an exception.

I am not sure why it works like that tho, we could easily fetch all collections and check their existence. 

But as the API does not return any value in case of remove operation we only check that sdk does not throw an exception.

Reply:
You are right - I was confused by the construction 

```csharp
var exception = await Record.ExceptionAsync(async () => await _client.ModifyCollectionAsync(changes));

// ...

Assert.Null(exception);

```